modules/appeals_api/app/controllers/appeals_api/legacy_appeals/v0/legacy_appeals_controller.rb

```diff
@@ -29,5 +26,16 @@ def schema
     def token_validation_api_key
       Settings.dig(:modules_appeals_api, :token_validation, :legacy_appeals, :api_key)
     end
+
+    def form_schema
+      @form_schema ||= AppealsApi::FormSchemas.new(
+        SCHEMA_ERROR_TYPE,
+        schema_version: 'v2'
+      )
+    end
+
+    def validate_json_schema_for_headers
+      form_schema.validate!(self.class::FORM_NUMBER, request_headers)
+    end
```

Comment:
This method overrides a method in the `AppealsApi::V2::DecisionReviews::LegacyAppealsController` so that the segmented endpoint uses the right schema (`LEGACY_APPEALS_HEADERS_WITH_SHARED_REFS`) for validation.